Where_Is_My_Wife/Assets/_Main/Scripts/Player/Controller/StateInterfaces.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using UniRx;
+using UnityEngine;
+
+namespace WhereIsMyWife.Controllers
+{
+    public interface IMovementStateEvents
+    {
+        IObservable<float> JumpStart { get; }
+        IObservable<float> Run { get; }
+        IObservable<float> GravityScale { get; }
+        IObservable<float> FallSpeedCap { get; }
+    }
+
+    public interface IDashStateEvents
+    {
+        IObservable<Vector2> Dash { get; }
+    }
+
+    public interface IHookStateEvents
+    {
+        
+    }
+
+    public interface IWallHangStateEvents
+    {
+        IObservable<Unit> StartWallHang { get; }
+        IObservable<float> WallHangVelocity { get; }
+        IObservable<float> WallJumpStart { get; }
+        IObservable<Unit> Turn { get; }
+    }
+    
+    public interface IWallJumpStateEvents
+    {
+        IObservable<float> WallJumpVelocity { get; }
+        IObservable<float> GravityScale { get; }
+        IObservable<float> FallSpeedCap { get; }
+    }
+}
```

Comment:
These can stay, we can use the same singletons to create different getters, instead of `Manager.Instance` use `Manager.WallJumpStateEvent => (IWallJumpStateEvents)Manager.Instance`

Reply:
da438a0813a32f4a61ffd9f71498940f5bb073a7